Sources/Shared/Core/BlueprintLayout.swift

```diff
@@ -156,37 +156,56 @@
     }
   }
 
-  /// Create supplementary layout attributes.
+  /// Resolve the size of SupplementaryView at index path.
+  /// If the collection view's delegate conforms to `(UI/NS)CollectionViewDelegateFlowLayout`, it will
+  /// query the delegate for the size of the SupplementaryView.
+  /// It defaults to using the `headerReferenceSize` or `footerReferenceSize` property on collection view flow layout.
   ///
-  /// - Parameters:
-  ///   - kind: The supplementary kind, either header or footer.
-  ///   - indexPath: The section index path for the supplementary view.
-  ///   - x: The x coordinate of the header layout attributes.
-  ///   - y: The y coordinate of the header layout attributes.
-  /// - Returns: A `LayoutAttributes` object of supplementary kind.
-  func createSupplementaryLayoutAttribute(ofKind kind: BlueprintSupplementaryKind,
-                                          indexPath: IndexPath,
-                                          atX x: CGFloat = 0,
-                                          atY y: CGFloat = 0) -> SupplementaryLayoutAttributes {
-    let layoutAttribute = SupplementaryLayoutAttributes(
-      forSupplementaryViewOfKind: kind.collectionViewSupplementaryType,
-      with: indexPath
-    )
-
+  /// - Parameter indexPath: The index path of the supplementaryView.
+  /// - Returns: The desired size of the item at the index path.
+  func resolveSizeForSupplementaryView(ofKind kind: BlueprintSupplementaryKind, at indexPath: IndexPath) -> CGSize {
     switch kind {
     case .header:
-      layoutAttribute.size.width = collectionView?.documentRect.width ?? headerReferenceSize.width
-      layoutAttribute.size.height = headerReferenceSize.height
+      let height = resolveCollectionView({ collectionView -> CGSize? in
+        return (collectionView.delegate as? CollectionViewFlowLayoutDelegate)?.collectionView?(collectionView,
+                                                                                               layout: self,
+                                                                                               referenceSizeForHeaderInSection: indexPath.section)
+      }, defaultValue: headerReferenceSize).height
+
+      let width: CGFloat
+      if headerReferenceSize.width > 0 {
+        width = headerReferenceSize.width
+      } else {
+        width = collectionView?.documentRect.width ?? headerReferenceSize.width
+      }
```

Comment:
Hmm that's a good point, I don't know why you would have it anything other than the collection view's width but it's one of the options for the delegate methods since you provide the size. 🤷‍♂️

Reply:
Perhaps the delegate should take precedence here if it exists. What do you think?

Reply:
Hmmm.. true but if the size is 0 it's not going to be visible hence why I took the collection views with before.

Reply:
But wouldn't you expect it to be zero if you delegate method returns zero?
I think it would be more frustrating if you try and set the width and you don't get the expected result.